vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/MobileAppWebDriverManager.java

```diff
@@ -120,21 +117,22 @@ private int getAndroidStatusBar()
 
     public double getDpr()
     {
-        return dpr.get();
-    }
-
-    private float calculateDpr()
-    {
-        byte[] imageBytes = getUnwrappedDriver(TakesScreenshot.class).getScreenshotAs(OutputType.BYTES);
-        try (InputStream is = new ByteArrayInputStream(imageBytes))
-        {
-            BufferedImage image = ImageIO.read(is);
-            return image.getHeight() / (float) getSize().getHeight();
-        }
-        catch (IOException e)
+        Double dpr = getWebDriverManagerContext().getParameter(WebDriverManagerParameter.DEVICE_PIXEL_RATIO);
```

Comment:
having this logic it makes sense introduce a new method in WebDriverManagerContext:
```java
<T> T get(WebDriverManagerParameter parameter, Supplier<T> initialValueSupplier);
```

Reply:
@valfirst are we going to remove initialValueSupplier from WebDriverManagerParameter? if yes we might miss default values for params that are assumed to alway have default non-null value (DESIRED_CAPABILITIES), otherwise imho having two suppliers present might make the API unclear

Reply:
> are we going to remove initialValueSupplier from WebDriverManagerParameter?

no, the proposed solution is temporal: we need to avoid duplication, since the logic is copy-pasted at least twice; actually we need 2 contexts: `WebDriverManagerContext` with *requested* parameters/capabilities/arguments and `WebDriverSession` with a set of runtime attributes that could be cached